targets/TARGET_Freescale/TARGET_KLXX/us_ticker.c

```diff
@@ -187,11 +187,11 @@ static void lptmr_isr(void) {
 void us_ticker_set_interrupt(timestamp_t timestamp) {
     int delta = (int)((uint32_t)timestamp - us_ticker_read());
     if (delta <= 0) {
-        // This event was in the past:
-        us_ticker_irq_handler();
-        return;
-    }
-    
+        // This event was in the past.  Force it into the very near
```

Comment:
This is a limitation of the API, and it's not related to the change in question, as that's the test the existing code uses.  (But for what it's worth, it's not valid in the API to create events more than 2^31-1 us in the future in the first place - this is mentioned in at least a few of the time-related APIs.)